jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfig.java

```diff
@@ -116,133 +111,50 @@ public boolean isRegistrySsl() {
-    // configure SDK metric exporter interval from jmx metric interval
-    getAndSetPropertyIfUndefined(properties, EXPORTER_INTERVAL, config.intervalMilliseconds);
-
-    config.metricsExporterType =
-        getAndSetPropertyIfUndefined(properties, METRICS_EXPORTER_TYPE, "logging");
-    if (OTLP_METRICS_EXPORTER.equalsIgnoreCase(config.metricsExporterType)) {
-      config.otlpExporterEndpoint =
-          getAndSetPropertyIfUndefined(properties, OTLP_ENDPOINT, "http://localhost:4318");
-    }
-    config.username = properties.getProperty(JMX_USERNAME);
-    config.password = properties.getProperty(JMX_PASSWORD);
-
-    config.remoteProfile = properties.getProperty(JMX_REMOTE_PROFILE);
-    config.realm = properties.getProperty(JMX_REALM);
-
-    config.registrySsl = Boolean.parseBoolean(properties.getProperty(REGISTRY_SSL));
-
-    validateConfig(config);
-    return config;
-  }
-
-  /**
-   * Sets system properties from effective configuration, must be called once and early before any
-   * OTel SDK or SSL/TLS stack initialization. This allows to override JVM system properties using
-   * user-provided configuration and also to set standard OTel SDK configuration.
-   */
-  public void propagateSystemProperties() {
-    for (Map.Entry<Object, Object> entry : properties.entrySet()) {
-
-      String key = entry.getKey().toString();
-      String value = entry.getValue().toString();
-      if (key.startsWith("otel.")
-          || key.startsWith("javax.net.ssl.keyStore")
-          || key.startsWith("javax.net.ssl.trustStore")) {
-        System.setProperty(key, value);
-      }
-    }
-  }
-
-  private static int getProperty(Properties properties, String key, int defaultValue)
-      throws ConfigurationException {
-    String propVal = properties.getProperty(key);
-    if (propVal == null) {
-      return defaultValue;
-    }
-    try {
-      return Integer.parseInt(propVal);
-    } catch (NumberFormatException e) {
-      throw new ConfigurationException("Failed to parse " + key, e);
-    }
-  }
-
-  /**
-   * Similar to getProperty(key, defaultValue) but sets the property to default if not in object.
    */
-  private static String getAndSetPropertyIfUndefined(
-      Properties properties, String key, String defaultValue) {
-    String propVal = properties.getProperty(key, defaultValue);
-    if (propVal.equals(defaultValue)) {
-      properties.setProperty(key, defaultValue);
+  public static JmxScraperConfig fromConfig(ConfigProperties config) {
+    JmxScraperConfig scraperConfig = new JmxScraperConfig();
+
+    Duration exportInterval = config.getDuration(METRIC_EXPORT_INTERVAL);
+    if (exportInterval == null || exportInterval.isNegative() || exportInterval.isZero()) {
+      // if not explicitly set, use default value of 1 minute as defined in specification
+      scraperConfig.samplingInterval = Duration.ofMinutes(1);
+    } else {
+      scraperConfig.samplingInterval = exportInterval;
```

Comment:
[for reviewer] we have to do this because we don't have access to the default value that will be used by the SDK when an explicit value is not present.